app/components/go_component.rb

```diff
@@ -0,0 +1,45 @@
+class GoComponent < ViewComponent::Base
+  include Motion::Component
+
+  map_motion :place
+  map_motion :reset
+
+  def initialize(key:)
+    @key = key
+    @game = GoGame.find(key: key) || GoGame.new(key: key)
+    puts "component game #{@game}"
+    @board = @game.display
+    @current = @game.current
+    @captures = @game.captures
+
+    stream_from "go:#{@key}", :player_move
+  end
+
+  def place(event)
+    index = event.target.data[:index].to_i
+    pos = GoGame::Pos.new((index / 9), index.modulo(9))
+
+    if @game.legal_move?(pos)
+      @game.place(pos)
+      @game.next_player
+      GoGame.update(key: @key, game: @game)
+    end
+  end
+
+  def reset
+    @game = GoGame.new
+    @board = @game.display
+    @current = @game.current
+    @captures = @game.captures
+    broadcast_board
+  end
+
+  private
+
+  def player_move(message)
+    @game = GoGame.find(key: @key)
```

Comment:
So I played with this a little bit.  There is so much going on with the internal state of the game!

The game doesn't work properly unless you both find the game and use the broadcast state for the board (at least).

I am concerned that what this shows is that there are two sources of truth:
a) the broadcast state of the board
b) the "saved" record of the game stored in the class variable

This might indeed be solved by storing this in a db.  But I wanted to point out that we'd probably prefer only one source of truth going forward.

Reply:
yeah i was a bit concerned about this. for some reason, if i didn't update the object in the `@game` variable in the component, a new object would get saved in the class variable. I'd hoped that if all components referred to the same object, i could achieve the "shared state". though this is probably "dangerous" because multiple clients could get in a bad state if they all try to make changes.

so this was the "solution" that seemed to work. updating the local variable to "re-fetch" the object stored in the class seemed to work.  probably i should update the component pieces then based on the game object and not use the broadcast message to send any state at all. just broadast something like "updated". 

i suppose this same thing would need to be used with the DB to ensure a component has an updated version of the record? I'm not sure how AR works if there's an object in memory that gets updated.